experimentation/tools/sorald/handled_rules.py

```diff
@@ -0,0 +1,203 @@
+import argparse
+import dataclasses
+import json
+import pathlib
+import re
+import subprocess
+import sys
+
+import jinja2
+
+from os.path import dirname, abspath, join
+from typing import Dict, List, Union
+
+from sorald._helpers import sonar_metadata, jsonkeys
+
+ENCODING = "utf8"
+
+BUG = []
+BUG_DETAIL = []
+
+CODE_SMELL = []
+CODE_SMELL_DETAIL = []
+
+VULNERABILITY = []
+VULNERABILITY_DETAIL = []
+
+TEMPLATE = r"""## Handled rules
+
+Sorald can currently repair violations of the following rules:
+
+* [Bug](#bug)
+{% for bug in bugs %}
+    * [{{ bug.title }}]({{ bug.link_to_detail }}) ([{{ bug.sonar_url_text }}]({{ bug.sonar_url }})) {% endfor %}
+
+* [Code Smell](#code-smell)
+{% for cs in code_smells %}
+    * [{{ cs.title }}]({{ cs.link_to_detail }}) ([{{ cs.sonar_url_text }}]({{ cs.sonar_url }})) {% endfor %}
+
+* [Vulnerability](#vulnerability)
+{% for vulnerability in vulnerabilities %}
+    * [{{ vulnerability.title }}]({{ vulnerability.link_to_detail }}) ([{{ vulnerability.sonar_url_text }}]({{ vulnerability.sonar_url }})) {% endfor %}
+
+### *Bug*
+{% for bug_content in bugs_detail %}
+#### {{ bug_content.title }} ([{{ bug_content.sonar_url_text }}]({{ bug_content.sonar_url }}))
+
+{{ bug_content.repair_description }}
+
+-----
+{% endfor %}
+### *Code Smell*
+{% for cs_content in code_smells_detail %}
+#### {{ cs_content.title }} ([{{ cs_content.sonar_url_text }}]({{ cs_content.sonar_url }}))
+
+{{ cs_content.repair_description }}
+
+-----
+{% endfor %}
+### *Vulnerability*
+{% for vulnerability_content in vulnerabilities_detail %}
+#### {{ vulnerability_content.title }} ([{{ vulnerability_content.sonar_url_text }}]({{ vulnerability_content.sonar_url }}))
+
+{{ vulnerability_content.repair_description }}
+
+-----
+{% endfor %}
+"""
+
+OUTPUT_ARG = "--output"
+
+PATH_TO_CLASSPATH = pathlib.Path(__file__).absolute().parent.parent.parent.parent / "cp.txt"
+with open(PATH_TO_CLASSPATH, 'r') as cp:
+    CLASSPATH = cp.read()
+
+PATH_TO_JAVA_SCRIPT = join(dirname(dirname(abspath(__file__))), "scripts", "GetKeyAndDescription.java")
```

Comment:
Using `pathlib` to resolve the path was throwing errors while executing the `JAVA` script. Somehow, the java CLI was interpreting `PosixPath`'s string representation a class and not a source file path. In other words, it was interpreting it as - `sorald.experimentation.tools.scripts.GetKeyAndDescription.java`

Reply:
This must be due to some other error on your end. If the Java CLI interpreted a string as dot-separated, that's because the string you passed to it was dot-separated.

Reply:
I will try to deal with this again and if I still face a problem, we can discuss this in the meeting tomorrow.

Anyway, we would no longer need this because we are removing the spoon script.